File: Downloads/mindboost_symfony/mindboost_symfony/src/Entity/GeneralAnswer.php
<?php

namespace App\Entity;

use App\Repository\GeneralAnswerRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: GeneralAnswerRepository::class)]
#[ORM\Table(name: 'general_answers')]
class GeneralAnswer
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: GeneralQuestion::class, inversedBy: 'answers')]
    #[ORM\JoinColumn(name: 'question_id', nullable: false, onDelete: 'CASCADE')]
    private ?GeneralQuestion $question = null;

    #[ORM\Column(name: 'answer_label', length: 10, options: ['default' => 'A'])]
    private string $answerLabel = 'A';

    #[ORM\Column(name: 'answer_text', length: 255)]
    #[Assert\NotBlank(message: 'Le texte de la réponse est obligatoire.')]
    private ?string $answerText = null;

    #[ORM\Column(type: 'integer', options: ['default' => 0])]
    #[Assert\Range(min: 0, max: 100)]
    private int $score = 0;

    #[ORM\Column(name: 'answer_order', type: 'integer')]
    private int $answerOrder = 1;

    #[ORM\Column(name: 'created_at', type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }
    public function getQuestion(): ?GeneralQuestion { return $this->question; }
    public function setQuestion(?GeneralQuestion $v): static { $this->question = $v; return $this; }
    public function getAnswerLabel(): string { return $this->answerLabel; }
    public function setAnswerLabel(string $v): static { $this->answerLabel = $v; return $this; }
    public function getAnswerText(): ?string { return $this->answerText; }
    public function setAnswerText(string $v): static { $this->answerText = $v; return $this; }
    public function getScore(): int { return $this->score; }
    public function setScore(int $v): static { $this->score = $v; return $this; }
    public function getAnswerOrder(): int { return $this->answerOrder; }
    public function setAnswerOrder(int $v): static { $this->answerOrder = $v; return $this; }
    public function getCreatedAt(): ?\DateTimeInterface { return $this->createdAt; }
    public function setCreatedAt(\DateTimeInterface $v): static { $this->createdAt = $v; return $this; }
    public function __toString(): string { return $this->answerText ?? ''; }
}

File: integrated-project/mindboost-javafx/src/main/java/com/mindboost/app/model/GeneralAnswer.java
package com.mindboost.app.model;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "general_answers")
public class GeneralAnswer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    private GeneralQuestion question;

    @Column(name = "answer_label", length = 10, nullable = false)
    private String answerLabel = "A";

    @Column(name = "answer_text", length = 255, nullable = false)
    private String answerText;

    @Column(nullable = false)
    private Integer score = 0;

    @Column(name = "answer_order", nullable = false)
    private Integer answerOrder = 1;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    public Integer getId() {
        return id;
    }

    public GeneralQuestion getQuestion() {
        return question;
    }

    public void setQuestion(GeneralQuestion question) {
        this.question = question;
    }

    public String getAnswerLabel() {
        return answerLabel;
    }

    public void setAnswerLabel(String answerLabel) {
        this.answerLabel = answerLabel;
    }

    public String getAnswerText() {
        return answerText;
    }

    public void setAnswerText(String answerText) {
        this.answerText = answerText;
    }

    public Integer getScore() {
        return score;
    }

    public void setScore(Integer score) {
        this.score = score;
    }

    public Integer getAnswerOrder() {
        return answerOrder;
    }

    public void setAnswerOrder(Integer answerOrder) {
        this.answerOrder = answerOrder;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}
